env/io_posix.cc

```diff
@@ -711,13 +711,19 @@ IOStatus PosixRandomAccessFile::MultiRead(FSReadRequest* reqs,
           // comment
           // https://github.com/facebook/rocksdb/pull/6441#issuecomment-589843435
           // Fall back to pread in this case.
-          Slice tmp_slice;
-          req->status =
-              Read(req->offset + req_wrap->finished_len,
-                   req->len - req_wrap->finished_len, options, &tmp_slice,
-                   req->scratch + req_wrap->finished_len, dbg);
-          req->result =
-              Slice(req->scratch, req_wrap->finished_len + tmp_slice.size());
+          if (use_direct_io()) {
+            // offset must be aligned, so read from the original offset again.
+            req->status = Read(req->offset, req->len, options, &req->result,
```

Comment:
With direct IO, I think this might be a more common occurrence, since if we're reading a block close to the end of the file, rounding up the size may extend the IO past the file. Why not just complete the request here instead of calling Read? Otherwise, we'll end up making 4 system calls overall for one IO, 2 in MultiRead and 2 in read.

Reply:
According to https://github.com/facebook/rocksdb/pull/6441#issuecomment-589843435, it might return partial result, even if the block is not near the end of file. So if we just complete the request, then we might have remaining bytes not read. If we can complete the request here, then logically we should be able to complete the request in non-direct IO mode too.

Reply:
Added a check for whether it's EOF.